com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalShaderGraphGUI.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.Rendering.HighDefinition;
+
+// Include material common properties names
+using static UnityEngine.Rendering.HighDefinition.HDMaterialProperties;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    /// <summary>
+    /// GUI for HDRP Shadergraph Decal materials.
+    /// </summary>
+    public class DecalShaderGraphGUI : HDShaderGUI
+    {
+        [Flags]
+        enum ExpandableBit : uint
+        {
+            SurfaceOptions = 1 << 0,
+            SurfaceInputs = 1 << 1,
+            Sorting = 1 << 2,
+        }
+
+        MaterialUIBlockList m_UIBlocks = new MaterialUIBlockList
+        {
+            new DecalSurfaceOptionsUIBlock((MaterialUIBlock.ExpandableBit)ExpandableBit.SurfaceOptions),
+            new ShaderGraphUIBlock((MaterialUIBlock.ExpandableBit)ExpandableBit.SurfaceInputs, ShaderGraphUIBlock.Features.None),
+            new DecalSortingInputsUIBlock((MaterialUIBlock.ExpandableBit)ExpandableBit.Sorting),
+        };
+
+        /// <summary>List of UI Blocks used to render the material inspector.</summary>
+        protected MaterialUIBlockList uiBlocks => m_UIBlocks;
+
+        /// <summary>
+        /// Implement your custom GUI in this function. To display a UI similar to HDRP shaders, use a MaterialUIBlockList.
```

Comment:
Override this function to implement your custom GUI. To display a user interface similar to HDRP shaders, use a MaterialUIBlockList.